xray/grpc.go

```diff
@@ -0,0 +1,174 @@
+package xray
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptrace"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	"go.uber.org/multierr"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/aws/aws-xray-sdk-go/header"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+// UnaryClientInterceptor provides gRPC unary client interceptor.
+func UnaryClientInterceptor(host string) grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		return Capture(ctx, host, func(ctx context.Context) error {
+			seg := GetSegment(ctx)
```

Comment:
This looks like it is getting the server segment and is then mutated below. The unit test seems sane though, could you point out where the subsegment is being created for the client request?

Reply:
Hello, @anuraaga
I believe the subsegment is created by `Capture` at line 26, which call `BeginSubsegment`.